File: Day 1/day1.cpp
#include <iostream>
#include <fstream>
#include <string>

int getLineSum(std::string const& line);

int main(int argc, char *argv[]) {
  if (argc < 2) {
    std::cout << "Did not provide file name!\n";
    return 0;
  }

  std::ifstream myFile;
  myFile.open(argv[1]);
  std::string s;
  int finalOutput = 0;

  if (myFile.is_open()) {
    while (myFile.good()) {
      myFile >> s;
      if (myFile.eof()) 
         break;
      finalOutput += getLineSum(s);
    }
  }

  std::cout << finalOutput << "\n";
  return 0;
}

int getLineSum(std::string const& line) {
  int first = -1;
  int last = -1;
  for (int i = 0; i < line.size(); i++) {
    if (line[i] >= 48 and line[i] <= 57) {
      if (first == -1) {
        first = line[i] - '0';
      }
      last = line[i] - '0';
    }
  }
  return first * 10 + last;
}

File: day1.py
filename = "day1.in"

def getLineSum(line):
    lp = 0
    rp = len(line) - 1
    first = -1
    last = -1
    while (first == -1):
        if (ord(line[lp]) >= 48 and ord(line[lp]) <= 57):
            first = int(line[lp])
        lp += 1
    while (last == -1):
        if (ord(line[rp]) >= 48 and ord(line[rp]) <= 57):
            last = int(line[rp])
        rp -= 1
    return first * 10 + last

with open(filename, "r") as f:
    finalSum = 0
    for line in f:
        finalSum += getLineSum(line)
    print(finalSum)